Update methods that check for widget presence

adsteel opened the pull request:
Addressing issue https://github.com/mojotech/dill/issues/83

adsteel commented:
Current plan is to:
- Formally deprecate `widget?`, `present?`, `absent?`
- Replace with `visible?` `not_visible?`
- Update the Readme

adsteel commented:
@david What do you think of this? I think maybe adding a `widget.visible?` method will be more complex, since we'll have to deal with the error raised for unfound widgets.

david commented:
@adsteel I think `not_visible?` will force people to wait, since it`s using `visible?` under the covers, which does wait.

adsteel commented:
@david Good point. I've switched it to use `has_no_selector`, which waits for the element to not be on the page. I've also added a second commit that swaps the deprecated methods for the new methods in tests.

aesnyder commented on lib/dill.rb:
```diff
@@ -20,4 +20,10 @@ class Missing < StandardError; end
   class AmbiguousWidget < StandardError; end
   class InvalidOption < StandardError; end
+
+  def deprecate(method, alternate_method, once=false)
```
nice

aesnyder commented on lib/dill/widgets/parts/container.rb:
```diff
@@ -1,12 +1,23 @@
 module Dill
   module WidgetParts
     module Container
+      include Dill
```
why the need to reinclude dill?

adsteel replied:
To allow access to the deprecated method.

aesnyder commented on lib/dill/widgets/parts/container.rb:
```diff
@@ -1,12 +1,23 @@
+      include Dill
+
       def has_widget?(name, *args)
+        deprecate('has_widget? and its alias widget?', 'visible?')
```
Is that to get access to the deprecate method here?

aesnyder commented:
:+1: LGTM
